### PR TITLE
Support the scope and variables requests

### PR DESCRIPTION
**Pull request by robertsipka**
Retrieve the list of variables at the current scope and show them under the variables pane.

This will be in WIP status until this [Jerryscript PR](https://github.com/jerryscript-project/jerryscript/pull/2557) does not landed.

### CODE REVIEW

**Comment by robertsipka** [comment]
The [Jerryscript PR](https://github.com/jerryscript-project/jerryscript/pull/2557) has just landed. I've removed the WIP flag.

**Review by LaszloLango — APPROVED** [review]
LGTM

File: src/IotjsDebugger.ts [hunk 1]
```diff
@@ -451,6 +452,55 @@ class IotjsDebugSession extends DebugSession {
...
   }
 
+
+  protected async scopesRequest(response: DebugProtocol.ScopesResponse, args: DebugProtocol.ScopesArguments
+  ): Promise<void> {
+    try {
+      const scopesArray: Array<JerryScopeChain> = await this._protocolhandler.requestScopes();
+      const scopes = new Array<Scope>();
+
+      for (let scope of scopesArray) {
```
Comment (rtakacs) [inline src/IotjsDebugger.ts]: Try to use `const` for the `for loop` variable.

File: src/IotjsDebugger.ts [hunk 2]
```diff
@@ -451,6 +452,55 @@ class IotjsDebugSession extends DebugSession {
...
+
+  protected async variablesRequest(response: DebugProtocol.VariablesResponse,
+                                   args: DebugProtocol.VariablesArguments
+  ): Promise<void> {
+    try {
+      const variables = new Array<DebugProtocol.Variable>();
+      const id = this._variableHandles.get(args.variablesReference);
+      const scopeVariables: Array<JerryScopeVariable> = await this._protocolhandler.requestVariables(Number(id));
+
+      for (let variable of scopeVariables) {
```
Comment (rtakacs) [inline src/IotjsDebugger.ts]: Ditto.

**Review by rtakacs — APPROVED** [review]
LGTM

**Pull request merged**